src/langchain_google_firestore/vectorstores.py

```diff
@@ -147,6 +152,66 @@ def add_texts(
 
         return _ids
 
+    def add_images(
+        self,
+        uris: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Adds image embeddings to Firestore vector store.
+
+        Args:
+            uris: A list of image uris (local, Google Cloud Storage or web)
+            metadatas: The metadata to add to the vector store. Defaults to None.
+            ids: The document ids to use for the new documents. If not provided, new
+            document ids will be generated.
+
+        Returns:
+            List[str]: The list of document ids added to the vector store.
+        """
+        images_len = len(list(uris))
+        ids_len_match = not ids or len(ids) == images_len
+        metadatas_len_match = not metadatas or len(metadatas) == images_len
+
+        if images_len == 0:
+            raise ValueError("No images provided to add to the vector store.")
+
+        if not metadatas_len_match:
+            raise ValueError(
+                "The length of metadatas must be the same as the length of images or zero."
+            )
+
+        if not ids_len_match:
+            raise ValueError(
+                "The length of ids must be the same as the length of images or zero."
+            )
+
+        if metadatas is None:
+            metadatas = [{"image_uri": uri} for uri in uris]
+
+        image_encodings = [self._encode_image(uri) for uri in uris]
+        image_embeddings = self._images_embedding_helper(list(uris))
```

Comment:
Correct me if I'm wrong, we still only support GCS and web urls through the VertexAIEmbeding service since other Embeddings services do not know how to handle GCS and Web uris?

Reply:
VertexAIEmbeddings supports GCS, Web, and local files. Other services probably only support web and local files